Broken build after updates to 2.6.1

After updates to 2.6.1 I start to recieve this error
```
node_modules/immutability-helper/index.d.ts(4,1): error TS2309: An export assignment cannot be used in a module with other exported elements.
node_modules/immutability-helper/index.d.ts(6,18): error TS2395: Individual declarations in merged declaration 'update' must be all exported or all local.
node_modules/immutability-helper/index.d.ts(8,18): error TS2395: Individual declarations in merged declaration 'update' must be all exported or all local.
node_modules/immutability-helper/index.d.ts(36,7): error TS2315: Type 'MapAndSetOperators' is not generic.

```
Any ideas how this could be fixed?

--- a/index.d.ts
+++ b/index.d.ts
@@ -6,26 +6,27 @@ export = update
 // declare function update<T>(data: T, query: update.Query<T>): T
 declare function update<T>(
   data: ReadonlyArray<T>,
-  query:
-    | {$push: T}
-    | {$unshift: T}
-    | {$splice: Array<[number, number]>}
-    | {$splice: Array<[number, number]>}
+  query: ArrayOperators<T>,
 ): ReadonlyArray<T>
 
 declare function update<T>(data: T, query: Query<T>): object
 declare function update<T>(
   data: ReadonlySet<T>,
-  query: {$add: any[]} | {$remove: string[]}
+  query: MapAndSetOperators<T>,
 ): ReadonlySet<T>
 
 declare function update<K, V>(
   data: ReadonlyMap<K, V>,
-  query: {$add: any[]} | {$remove: string[]}
+  query: MapAndSetOperators<T>,
 ): ReadonlyMap<K, V>
 
 type Tree<T> = {[K in keyof T]?: Query<T[K]>}
-type Query<T> = Tree<T> | ObjectOperators<T>
+type Query<T> =
+  | Tree<T>
+  | ObjectOperators<T>
+  | ArrayOperators<T>
+  | MapAndSetOperators<T>
+
 type ObjectOperators<T> =
   | {$set: any}
   | {$toggle: Array<keyof T>}
@@ -33,12 +34,19 @@ type ObjectOperators<T> =
   | {$merge: Partial<T>}
   | {$apply: (old: T) => any}
   | ((old: T) => any)
+type ArrayOperators<T> =
+  | {$push: T}
+  | {$unshift: T}
+  | {$splice: Array<[number, number]>}
+  | {$splice: Array<[number, number]>}
+
+type MapAndSetOperators<T> = {$add: any[]} | {$remove: string[]}
 
 declare namespace update {
   function newContext(): typeof update
   function extend<T>(
     command: Command,
-    handler: (param: CommandArg, old: T) => T
+    handler: (param: CommandArg, old: T) => T,
   ): void
 
   type Command = string

--- a/index.d.ts
+++ b/index.d.ts
@@ -12,12 +12,12 @@ declare function update<T>(
 declare function update<T>(data: T, query: Query<T>): object
 declare function update<T>(
   data: ReadonlySet<T>,
-  query: MapAndSetOperators<T>,
+  query: SetOperators<T>,
 ): ReadonlySet<T>
 
 declare function update<K, V>(
   data: ReadonlyMap<K, V>,
-  query: MapAndSetOperators<T>,
+  query: MapOperators<K, V>,
 ): ReadonlyMap<K, V>
 
 type Tree<T> = {[K in keyof T]?: Query<T[K]>}
@@ -25,7 +25,7 @@ type Query<T> =
   | Tree<T>
   | ObjectOperators<T>
   | ArrayOperators<T>
-  | MapAndSetOperators<T>
+  | SetOperators<T>
 
 type ObjectOperators<T> =
   | {$set: any}
@@ -40,7 +40,8 @@ type ArrayOperators<T> =
   | {$splice: Array<[number, number]>}
   | {$splice: Array<[number, number]>}
 
-type MapAndSetOperators<T> = {$add: any[]} | {$remove: string[]}
+type MapOperators<K, V> = {$add: Array<[K, V]>} | {$remove: K[]}
+type SetOperators<T> = {$add: T[]} | {$remove: T[]}
 
 declare namespace update {
   function newContext(): typeof update

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,0 +1,41 @@
+// Type definitions for immutability-helper
+// Project: Immutability helper
+
+export default update
+
+declare function update<T>(data: T, query: update.Query<T>): T
+
+declare namespace update {
+  function newContext(): typeof update
+  function extend<T>(
+    command: Command,
+    handler: (param: CommandArg, old: T) => T
+  )
+
+  type Command = string
+  type CommandArg = any
+
+  type ObjectOperator<T> =
+    | {$set: any}
+    | {$toggle: Array<keyof T>}
+    | {$unset: Array<keyof T>}
+    | {$merge: Partial<T>}
+    | {$apply: (old: T) => any}
+    | ((old: T) => any)
+
+  type MapAndSetOperators<T> = {$add: Array<any>} | {$remove: Array<string>}
+
+  type ArrayOperator<T> =
+    | {$push: T}
+    | {$unshift: T}
+    | {$splice: [number, number][]}
+
+  type Operators<T> =
+    | ObjectOperator<T>
+    | ArrayOperator<T>
+    | MapAndSetOperators<T>
+
+  type Tree<T> = {[K in keyof T]?: Query<T[K]>}
+
+  type Query<T> = Tree<T> | Operators<T>
+}

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,41 +1,46 @@
-// Type definitions for immutability-helper
 // Project: Immutability helper
+// TypeScript Version: 2.2
 
-export default update
+export = update
 
-declare function update<T>(data: T, query: update.Query<T>): T
+// declare function update<T>(data: T, query: update.Query<T>): T
+declare function update<T>(
+  data: ReadonlyArray<T>,
+  query:
+    | {$push: T}
+    | {$unshift: T}
+    | {$splice: Array<[number, number]>}
+    | {$splice: Array<[number, number]>}
+): ReadonlyArray<T>
+
+declare function update<T>(data: T, query: Query<T>): object
+declare function update<T>(
+  data: ReadonlySet<T>,
+  query: {$add: any[]} | {$remove: string[]}
+): ReadonlySet<T>
+
+declare function update<K, V>(
+  data: ReadonlyMap<K, V>,
+  query: {$add: any[]} | {$remove: string[]}
+): ReadonlyMap<K, V>
+
+type Tree<T> = {[K in keyof T]?: Query<T[K]>}
+type Query<T> = Tree<T> | ObjectOperators<T>
+type ObjectOperators<T> =
+  | {$set: any}
+  | {$toggle: Array<keyof T>}
+  | {$unset: Array<keyof T>}
+  | {$merge: Partial<T>}
+  | {$apply: (old: T) => any}
+  | ((old: T) => any)
 
 declare namespace update {
   function newContext(): typeof update
   function extend<T>(
     command: Command,
     handler: (param: CommandArg, old: T) => T
-  )
+  ): void
 
   type Command = string
   type CommandArg = any
-
-  type ObjectOperator<T> =
-    | {$set: any}
-    | {$toggle: Array<keyof T>}
-    | {$unset: Array<keyof T>}
-    | {$merge: Partial<T>}
-    | {$apply: (old: T) => any}
-    | ((old: T) => any)
-
-  type MapAndSetOperators<T> = {$add: Array<any>} | {$remove: Array<string>}
-
-  type ArrayOperator<T> =
-    | {$push: T}
-    | {$unshift: T}
-    | {$splice: [number, number][]}
-
-  type Operators<T> =
-    | ObjectOperator<T>
-    | ArrayOperator<T>
-    | MapAndSetOperators<T>
-
-  type Tree<T> = {[K in keyof T]?: Query<T[K]>}
-
-  type Query<T> = Tree<T> | Operators<T>
 }

--- a/index.d.ts
+++ b/index.d.ts
@@ -24,8 +24,8 @@ type Tree<T> = {[K in keyof T]?: Query<T[K]>}
 type Query<T> =
   | Tree<T>
   | ObjectOperators<T>
-  | ArrayOperators<T>
-  | SetOperators<T>
+  | ArrayOperators<any>
+  | SetOperators<any>
 
 type ObjectOperators<T> =
   | {$set: any}

--- a/index.d.ts
+++ b/index.d.ts
@@ -38,7 +38,6 @@ type ArrayOperators<T> =
   | {$push: T}
   | {$unshift: T}
   | {$splice: Array<[number, number]>}
-  | {$splice: Array<[number, number]>}
 
 type MapOperators<K, V> = {$add: Array<[K, V]>} | {$remove: K[]}
 type SetOperators<T> = {$add: T[]} | {$remove: T[]}

--- a/index.d.ts
+++ b/index.d.ts
@@ -9,7 +9,6 @@ declare function update<T>(
   query: ArrayOperators<T>,
 ): ReadonlyArray<T>
 
-declare function update<T>(data: T, query: Query<T>): object
 declare function update<T>(
   data: ReadonlySet<T>,
   query: SetOperators<T>,
@@ -19,6 +18,8 @@ declare function update<K, V>(
   data: ReadonlyMap<K, V>,
   query: MapOperators<K, V>,
 ): ReadonlyMap<K, V>
+
+declare function update<T>(data: T, query: Query<T>): object
 
 type Tree<T> = {[K in keyof T]?: Query<T[K]>}
 type Query<T> =
@@ -32,7 +33,7 @@ type ObjectOperators<T> =
   | {$toggle: Array<keyof T>}
   | {$unset: Array<keyof T>}
   | {$merge: Partial<T>}
-  | {$apply: (old: T) => any}
+  | {$apply: (old: T) => T}
   | ((old: T) => any)
 type ArrayOperators<T> =
   | {$push: T}

--- a/package.json
+++ b/package.json
@@ -7,7 +7,8 @@
   "scripts": {
     "test-cov": "nyc npm test && nyc report --reporter=lcov",
     "test-travis": "nyc npm test && nyc report --reporter=lcov",
-    "test": "mocha test.js"
+    "test": "mocha test.js",
+    "test-dtslint": "dtslint"
   },
   "keywords": [
     "immutability"
@@ -16,6 +17,7 @@
   "license": "MIT",
   "devDependencies": {
     "coveralls": "^2.11.6",
+    "dtslint": "^0.2.0",
     "expect": "^1.14.0",
     "mocha": "^2.4.5",
     "nyc": "^5.6.0"

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,0 +1,15 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "lib": ["es6"],
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "strictNullChecks": true,
+    "noEmit": true,
+    "strictFunctionTypes": false,
+
+    "baseUrl": ".",
+    "paths": { "immutability-helper": ["."] }
+  }
+}
+

--- a/tslint.json
+++ b/tslint.json
@@ -1,0 +1,7 @@
+{
+    "extends": "dtslint/dtslint.json",
+    "rules": {
+        "semicolon": false
+    }
+}
+

